tests/e2e/isbservice.go

```diff
@@ -229,3 +232,145 @@ func startISBServiceRolloutWatches() {
 	wg.Add(1)
 	go watchStatefulSet()
 }
+
+// shared functions
+
+// create an ISBServiceRollout of a given version and name and make sure it's running
+func createISBServiceRollout(name string, isbServiceSpec numaflowv1.InterStepBufferServiceSpec) {
+
+	isbServiceRolloutSpec := createISBServiceRolloutSpec(name, Namespace, isbServiceSpec)
+	_, err := isbServiceRolloutClient.Create(ctx, isbServiceRolloutSpec, metav1.CreateOptions{})
+	Expect(err).ShouldNot(HaveOccurred())
+
+	document("Verifying that the ISBServiceRollout was created")
+	Eventually(func() error {
+		_, err := isbServiceRolloutClient.Get(ctx, name, metav1.GetOptions{})
+		return err
+	}, testTimeout, testPollingInterval).Should(Succeed())
+
+	verifyISBSvcRolloutReady(name)
+
+	verifyISBSvcReady(Namespace, name, 3)
+
+}
+
+func createISBServiceRolloutSpec(name, namespace string, isbServiceSpec numaflowv1.InterStepBufferServiceSpec) *apiv1.ISBServiceRollout {
+
+	rawSpec, err := json.Marshal(isbServiceSpec)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	isbServiceRollout := &apiv1.ISBServiceRollout{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "numaplane.numaproj.io/v1alpha1",
+			Kind:       "ISBServiceRollout",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: apiv1.ISBServiceRolloutSpec{
+			InterStepBufferService: apiv1.InterStepBufferService{
+				Spec: runtime.RawExtension{
+					Raw: rawSpec,
+				},
+			},
+		},
+	}
+
+	return isbServiceRollout
+
+}
+
+// delete ISBServiceRollout and verify deletion
+func deleteISBServiceRollout(name string) {
+	document("Deleting ISBServiceRollout")
+	err := isbServiceRolloutClient.Delete(ctx, name, metav1.DeleteOptions{})
+	Expect(err).ShouldNot(HaveOccurred())
+
+	document("Verifying ISBServiceRollout deletion")
+	Eventually(func() bool {
+		_, err := isbServiceRolloutClient.Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				Fail("An unexpected error occurred when fetching the ISBServiceRollout: " + err.Error())
+			}
+			return false
+		}
+		return true
+	}).WithTimeout(testTimeout).Should(BeFalse(), "The ISBServiceRollout should have been deleted but it was found.")
+
+	document("Verifying ISBService deletion")
+	Eventually(func() bool {
+		list, err := dynamicClient.Resource(getGVRForISBService()).Namespace(Namespace).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return false
+		}
+		if len(list.Items) == 0 {
+			return true
+		}
+		return false
+	}).WithTimeout(testTimeout).Should(BeTrue(), "The ISBService should have been deleted but it was found.")
+}
+
+func updateISBServiceRollout(isbServiceRolloutName, pipelineRolloutName string, newSpec numaflowv1.InterStepBufferServiceSpec, f func(numaflowv1.InterStepBufferServiceSpec) bool, dataLoss bool) {
```

Comment:
nit: can you change the name `f` to something like `verifySpecFunc`?

Reply:
Actually, maybe it's fine. In that case, maybe just comment the function to explain what the parameters mean?

Reply:
Can you add a TODO to make `pipelineRolloutName` an array?